### ISSUE TITLE
Extension, Static Libraries, Sharing pods

### ISSUE THREAD

**Issue by radvansky-tomas** [CocoaPods/CocoaPods#8321]
Hi, I am trying to use cocoa pods in slightly complex application, but it seems that libraries I am trying to use are not really compatible with such architecture.

So I would like to ask you to help me configure cocoa pods to make it possible:
I have main app, shared framework, watch app and today extension.

Shared framework contains most of the logic, model, actions etc. Watch only use that framework via network communication via host app. But today extension is separate app bundled with rest of targets.

Problem becomes obvious once I am trying to add dependencies, mainly Firebase SDKs and Swifter Swift. (but others as well)

Libraries got linked multiple times, causing problems with duplicate errors (and crashes). Another issue is that some libraries are built with "Allow only extension API", which for some of them is fatal and they will not build at all.

I tried to create separate private POD for my shared framework and link libraries outside of main project, but Firebase doesnt like to be encapsulated again, I had to use static framework = true and again other things wont build...

So basically my question is, can I configure those targets as completely isolated project, with own dependencies ? I saw flag :exclusive for project but I am getting Unsupported options `{:exclusive=>true}` for target `today-dev` So I guess it was deprecated...

**Comment by dnkoutso** [comment]
@radvansky-tomas it is not exactly easy with the given description and no sample app for anyone to assist.

Furthermore, it is hard to also derive if there is an issue in CocoaPods library without a sample app demonstrating the issue.

I apologize but I am going to have to close this issue (we can continue the discussion and I can-reopen if we find a bug). 

I also recommend you search here for additional issues if there is a duplicate and additionally file an issue on StackOverflow under cocoapods tag.

**Comment by radvansky-tomas** [comment]
This is probably one the most relevant I found, tried and used https://github.com/CocoaPods/CocoaPods/issues/5768

**Comment by radvansky-tomas** [comment]
I will create sample project demonstrating the issue

**Comment by radvansky-tomas** [comment]
Another question is about module name: For example swifter swift library. https://github.com/SwifterSwift/SwifterSwift contains subspecs, and I can reference supported ones to main app and extensions one to extension target. BUT cocoa pods then creates 2 frameworks, with 2 wrong names, and build fails, as module name is unknown. 

![image](https://user-images.githubusercontent.com/12473664/49350891-c7027100-f715-11e8-9061-d22ea5dc3675.png)

**Comment by dnkoutso** [comment]
yes that is because two different targets use two different specs (including subspecs) for a pod.

**Comment by dnkoutso** [comment]
There might be a bug there definitely, would love sample app and I will re-open it.

**Comment by a83988029** [comment]
run into this bugs too
